### PR TITLE
Load incomplete payments on startup

### LINKED ISSUE #266: Load incomplete payments on startup

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #266

### CODE REVIEW

File: golem/ethereum/paymentprocessor.py [hunk 1]
```diff
@@ -166,9 +168,18 @@ def _gnt_reserved(self):
...
+        with db.atomic():
+            for awaiting_payment in Payment\
+                    .select()\
+                    .where(Payment.status == PaymentStatus.awaiting):
+                self.add(awaiting_payment)
+
     def add(self, payment, deadline=DEFAULT_DEADLINE):
         if payment.status is not PaymentStatus.awaiting:
             raise RuntimeError("Invalid payment status: {}".format(payment.status))
+        if payment in self._awaiting:
```
Comment (chfast) [inline golem/ethereum/paymentprocessor.py]: When this can happen?

**Review by chfast — COMMENTED** [review]
Looks good apart one concern raised.

But in general we should load all non-confirmed payments, not only awaiting?

File: golem/ethereum/paymentprocessor.py (hunk 1 above)
Comment (jiivan) [reply golem/ethereum/paymentprocessor.py]: Hopefully it won't happen, but IMHO it's an easy check, worth adding.

Comment (chfast) [reply golem/ethereum/paymentprocessor.py]: For me it is exactly the opposite. The defensive code like this usually hides problems and means we have not idea what our code is doing. I can accept exception here, but not silent break.

**Review by chfast — CHANGES_REQUESTED** [review]

**Review by chfast — APPROVED** [review]

**Comment by chfast** [comment]
:shipit:

**Pull request merged**